plugins/aja/meson.build

```diff
@@ -0,0 +1,47 @@
+add_languages('cpp', required: true)
+
+# Find NTV2 libraries
+ajantv2_dep = cc.find_library('ajantv2d', required: false, dirs: [ obs_libdir ])
+if not ajantv2_dep.found()
+  ajantv2_dep = cc.find_library('ajantv2', dirs: [ obs_libdir ])
+endif
+
+aja_deps = [
+  ajantv2_dep,
+  libobs_dep,
+]
+
+aja_include_directories = [
+  include_directories(obs_includedir / 'ajalibraries'),
+]
+
+aja_sources = files([
+  'aja-card-manager.cpp',
+  'aja-common.cpp',
+  'aja-output.cpp',
+  'aja-presets.cpp',
+  'aja-props.cpp',
+  'aja-routing.cpp',
+  'aja-source.cpp',
+  'aja-vpid-data.cpp',
+  'aja-widget-io.cpp',
+  'main.cpp',
+])
+
+aja_cpp_args = []
+if host_system == 'linux'
+  aja_cpp_args += [ '-DAJALinux', '-DAJA_LINUX' ]
+endif
+
+aja_plugin = shared_library('aja',
```

Comment:
In all these plugins, you use `shared_library()`. There's a `shared_module()` function that specifically creates libraries useful for `dlopen()`, that cannot be directly linked to, for example they don't get a soname (Debian dpkg-shlibdeps distinguishes between libraries with and without soname), and they are built without making undefined symbols an error since those are likely provided by the environment that dlopened it.

Reply:
Opened https://github.com/GeorgesStavracas/obs-gtk/pull/5 for that